Mono: Can't convert type Godot.Node to T

**Godot version:**
Master


**OS/device including version:**
Windows 10


**Issue description:**

<details>
<summary>**Expand to see Full Log**</summary>

```
Project "GodotSharp.sln" (Build target(s)):
	Message: Building solution configuration "Release|Any CPU".
	Project "GodotSharp.csproj" (default targets):
		Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
		Csc: D:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Roslyn\csc.exe /noconfig /nowarn:1591,1701,1702 /nostdlib+ /errorreport:prompt /warn:4 /doc:bin\Release\GodotSharp.xml /highentropyva+ /reference:"C:\Program Files\Mono\lib\mono\4.5\mscorlib.dll" /reference:"C:\Program Files\Mono\lib\mono\4.5\System.Core.dll" /reference:"C:\Program Files\Mono\lib\mono\4.5\System.dll" /debug:portable /optimize+ /out:obj\Release\GodotSharp.dll /ruleset:"D:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Team Tools\Static Analysis Tools\\Rule Sets\MinimumRecommendedRules.ruleset" /subsystemversion:6.00 /target:library /utf8output Core\AABB.cs Core\Array.cs Core\AttributesExportAttribute.cs Core\AttributesGodotMethodAttribute.cs Core\AttributesRPCAttributes.cs Core\AttributesSignalAttribute.cs Core\AttributesToolAttribute.cs Core\Basis.cs Core\Color.cs Core\DebuggingUtils.cs Core\Dictionary.cs Core\ExtensionsNodeExtensions.cs Core\ExtensionsObjectExtensions.cs Core\ExtensionsResourceLoaderExtensions.cs Core\GD.cs Core\GD_constants.cs Core\GodotSynchronizationContext.cs Core\GodotTaskScheduler.cs Core\InterfacesIAwaitable.cs Core\InterfacesIAwaiter.cs Core\MarshalUtils.cs Core\Mathf.cs Core\MathfEx.cs Core\NativeCalls.cs Core\NodePath.cs Core\Object.base.cs Core\Plane.cs Core\Quat.cs Core\Rect2.cs Core\RID.cs Core\SignalAwaiter.cs Core\StringExtensions.cs Core\Transform.cs Core\Transform2D.cs Core\Vector2.cs Core\Vector3.cs ObjectType\AcceptDialog.cs ObjectType\AnimatedSprite.cs ObjectType\AnimatedSprite3D.cs ObjectType\AnimatedTexture.cs ObjectType\Animation.cs ObjectType\AnimationNode.cs ObjectType\AnimationNodeAdd2.cs ObjectType\AnimationNodeAdd3.cs ObjectType\AnimationNodeAnimation.cs ObjectType\AnimationNodeBlend2.cs ObjectType\AnimationNodeBlend3.cs ObjectType\AnimationNodeBlendSpace1D.cs ObjectType\AnimationNodeBlendSpace2D.cs ObjectType\AnimationNodeBlendTree.cs ObjectType\AnimationNodeOneShot.cs ObjectType\AnimationNodeOutput.cs ObjectType\AnimationNodeStateMachine.cs ObjectType\AnimationNodeStateMachinePlayback.cs ObjectType\AnimationNodeStateMachineTransition.cs ObjectType\AnimationNodeTimeScale.cs ObjectType\AnimationNodeTimeSeek.cs ObjectType\AnimationNodeTransition.cs ObjectType\AnimationPlayer.cs ObjectType\AnimationRootNode.cs ObjectType\AnimationTree.cs ObjectType\AnimationTreePlayer.cs ObjectType\Area.cs ObjectType\Area2D.cs ObjectType\ArrayMesh.cs ObjectType\ARVRAnchor.cs ObjectType\ARVRCamera.cs ObjectType\ARVRController.cs ObjectType\ARVRInterface.cs ObjectType\ARVRInterfaceGDNative.cs ObjectType\ARVROrigin.cs ObjectType\ARVRPositionalTracker.cs ObjectType\ARVRServer.cs ObjectType\AStar.cs ObjectType\AtlasTexture.cs ObjectType\AudioBusLayout.cs ObjectType\AudioEffect.cs ObjectType\AudioEffectAmplify.cs ObjectType\AudioEffectBandLimitFilter.cs ObjectType\AudioEffectBandPassFilter.cs ObjectType\AudioEffectChorus.cs ObjectType\AudioEffectCompressor.cs ObjectType\AudioEffectDelay.cs ObjectType\AudioEffectDistortion.cs ObjectType\AudioEffectEQ.cs ObjectType\AudioEffectEQ10.cs ObjectType\AudioEffectEQ21.cs ObjectType\AudioEffectEQ6.cs ObjectType\AudioEffectFilter.cs ObjectType\AudioEffectHighPassFilter.cs ObjectType\AudioEffectHighShelfFilter.cs ObjectType\AudioEffectLimiter.cs ObjectType\AudioEffectLowPassFilter.cs ObjectType\AudioEffectLowShelfFilter.cs ObjectType\AudioEffectNotchFilter.cs ObjectType\AudioEffectPanner.cs ObjectType\AudioEffectPhaser.cs ObjectType\AudioEffectPitchShift.cs ObjectType\AudioEffectRecord.cs ObjectType\AudioEffectReverb.cs ObjectType\AudioEffectStereoEnhance.cs ObjectType\AudioServer.cs ObjectType\AudioStream.cs ObjectType\AudioStreamMicrophone.cs ObjectType\AudioStreamOGGVorbis.cs ObjectType\AudioStreamPlayback.cs ObjectType\AudioStreamPlayer.cs ObjectType\AudioStreamPlayer2D.cs ObjectType\AudioStreamPlayer3D.cs ObjectType\AudioStreamRandomPitch.cs ObjectType\AudioStreamSample.cs ObjectType\BackBufferCopy.cs ObjectType\BakedLightmap.cs ObjectType\BakedLightmapData.cs ObjectType\BaseButton.cs ObjectType\BitMap.cs ObjectType\BitmapFont.cs ObjectType\Bone2D.cs ObjectType\BoneAttachment.cs ObjectType\BoxContainer.cs ObjectType\BoxShape.cs ObjectType\Button.cs ObjectType\ButtonGroup.cs ObjectType\Camera.cs ObjectType\Camera2D.cs ObjectType\CanvasItem.cs ObjectType\CanvasItemMaterial.cs ObjectType\CanvasLayer.cs ObjectType\CanvasModulate.cs ObjectType\CapsuleMesh.cs ObjectType\CapsuleShape.cs ObjectType\CapsuleShape2D.cs ObjectType\CenterContainer.cs ObjectType\CheckBox.cs ObjectType\CheckButton.cs ObjectType\CircleShape2D.cs ObjectType\ClassDB.cs ObjectType\ClippedCamera.cs ObjectType\CollisionObject.cs ObjectType\CollisionObject2D.cs ObjectType\CollisionPolygon.cs ObjectType\CollisionPolygon2D.cs ObjectType\CollisionShape.cs ObjectType\CollisionShape2D.cs ObjectType\ColorPicker.cs ObjectType\ColorPickerButton.cs ObjectType\ColorRect.cs ObjectType\ConcavePolygonShape.cs ObjectType\ConcavePolygonShape2D.cs ObjectType\ConeTwistJoint.cs ObjectType\ConfigFile.cs ObjectType\ConfirmationDialog.cs ObjectType\Container.cs ObjectType\Control.cs ObjectType\ConvexPolygonShape.cs ObjectType\ConvexPolygonShape2D.cs ObjectType\CPUParticles.cs ObjectType\CPUParticles2D.cs ObjectType\CSGBox.cs ObjectType\CSGCombiner.cs ObjectType\CSGCylinder.cs ObjectType\CSGMesh.cs ObjectType\CSGPolygon.cs ObjectType\CSGPrimitive.cs ObjectType\CSGShape.cs ObjectType\CSGSphere.cs ObjectType\CSGTorus.cs ObjectType\CSharpScript.cs ObjectType\CubeMap.cs ObjectType\CubeMesh.cs ObjectType\Curve.cs ObjectType\Curve2D.cs ObjectType\Curve3D.cs ObjectType\CurveTexture.cs ObjectType\CylinderMesh.cs ObjectType\CylinderShape.cs ObjectType\DampedSpringJoint2D.cs ObjectType\DirectionalLight.cs ObjectType\Directory.cs ObjectType\DynamicFont.cs ObjectType\DynamicFontData.cs ObjectType\EncodedObjectAsID.cs ObjectType\Engine.cs ObjectType\Environment.cs ObjectType\Expression.cs ObjectType\File.cs ObjectType\FileDialog.cs ObjectType\Font.cs ObjectType\FuncRef.cs ObjectType\GDNative.cs ObjectType\GDNativeLibrary.cs ObjectType\GDScript.cs ObjectType\GDScriptFunctionState.cs ObjectType\Generic6DOFJoint.cs ObjectType\Geometry.cs ObjectType\GeometryInstance.cs ObjectType\GIProbe.cs ObjectType\GIProbeData.cs ObjectType\GodotSharp.cs ObjectType\Gradient.cs ObjectType\GradientTexture.cs ObjectType\GraphEdit.cs ObjectType\GraphNode.cs ObjectType\GridContainer.cs ObjectType\GridMap.cs ObjectType\GrooveJoint2D.cs ObjectType\HBoxContainer.cs ObjectType\HingeJoint.cs ObjectType\HScrollBar.cs ObjectType\HSeparator.cs ObjectType\HSlider.cs ObjectType\HSplitContainer.cs ObjectType\HTTPClient.cs ObjectType\HTTPRequest.cs ObjectType\Image.cs ObjectType\ImageTexture.cs ObjectType\ImmediateGeometry.cs ObjectType\Input.cs ObjectType\InputEvent.cs ObjectType\InputEventAction.cs ObjectType\InputEventGesture.cs ObjectType\InputEventJoypadButton.cs ObjectType\InputEventJoypadMotion.cs ObjectType\InputEventKey.cs ObjectType\InputEventMagnifyGesture.cs ObjectType\InputEventMouse.cs ObjectType\InputEventMouseButton.cs ObjectType\InputEventMouseMotion.cs ObjectType\InputEventPanGesture.cs ObjectType\InputEventScreenDrag.cs ObjectType\InputEventScreenTouch.cs ObjectType\InputEventWithModifiers.cs ObjectType\InputMap.cs ObjectType\InstancePlaceholder.cs ObjectType\InterpolatedCamera.cs ObjectType\IP.cs ObjectType\ItemList.cs ObjectType\JavaScript.cs ObjectType\Joint.cs ObjectType\Joint2D.cs ObjectType\JSON.cs ObjectType\JSONParseResult.cs ObjectType\KinematicBody.cs ObjectType\KinematicBody2D.cs ObjectType\KinematicCollision.cs ObjectType\KinematicCollision2D.cs ObjectType\Label.cs ObjectType\LargeTexture.cs ObjectType\Light.cs ObjectType\Light2D.cs ObjectType\LightOccluder2D.cs ObjectType\Line2D.cs ObjectType\LineEdit.cs ObjectType\LineShape2D.cs ObjectType\LinkButton.cs ObjectType\Listener.cs ObjectType\MainLoop.cs ObjectType\MarginContainer.cs ObjectType\Marshalls.cs ObjectType\Material.cs ObjectType\MenuButton.cs ObjectType\Mesh.cs ObjectType\MeshDataTool.cs ObjectType\MeshInstance.cs ObjectType\MeshInstance2D.cs ObjectType\MeshLibrary.cs ObjectType\MobileVRInterface.cs ObjectType\MultiMesh.cs ObjectType\MultiMeshInstance.cs ObjectType\MultiplayerAPI.cs ObjectType\MultiplayerPeerGDNative.cs ObjectType\Mutex.cs ObjectType\NativeScript.cs ObjectType\Navigation.cs ObjectType\Navigation2D.cs ObjectType\NavigationMesh.cs ObjectType\NavigationMeshInstance.cs ObjectType\NavigationPolygon.cs ObjectType\NavigationPolygonInstance.cs ObjectType\NetworkedMultiplayerENet.cs ObjectType\NetworkedMultiplayerPeer.cs ObjectType\NinePatchRect.cs ObjectType\Node.cs ObjectType\Node2D.cs ObjectType\NoiseTexture.cs ObjectType\Object.cs ObjectType\OccluderPolygon2D.cs ObjectType\OmniLight.cs ObjectType\OpenSimplexNoise.cs ObjectType\OptionButton.cs ObjectType\OrientedPathFollow.cs ObjectType\OS.cs ObjectType\PackedDataContainer.cs ObjectType\PackedDataContainerRef.cs ObjectType\PackedScene.cs ObjectType\PacketPeer.cs ObjectType\PacketPeerGDNative.cs ObjectType\PacketPeerStream.cs ObjectType\PacketPeerUDP.cs ObjectType\Panel.cs ObjectType\PanelContainer.cs ObjectType\PanoramaSky.cs ObjectType\ParallaxBackground.cs ObjectType\ParallaxLayer.cs ObjectType\Particles.cs ObjectType\Particles2D.cs ObjectType\ParticlesMaterial.cs ObjectType\Path.cs ObjectType\Path2D.cs ObjectType\PathFollow.cs ObjectType\PathFollow2D.cs ObjectType\PCKPacker.cs ObjectType\Performance.cs ObjectType\PHashTranslation.cs ObjectType\PhysicalBone.cs ObjectType\Physics2DDirectBodyState.cs ObjectType\Physics2DDirectSpaceState.cs ObjectType\Physics2DServer.cs ObjectType\Physics2DShapeQueryParameters.cs ObjectType\Physics2DShapeQueryResult.cs ObjectType\Physics2DTestMotionResult.cs ObjectType\PhysicsBody.cs ObjectType\PhysicsBody2D.cs ObjectType\PhysicsDirectBodyState.cs ObjectType\PhysicsDirectSpaceState.cs ObjectType\PhysicsMaterial.cs ObjectType\PhysicsServer.cs ObjectType\PhysicsShapeQueryParameters.cs ObjectType\PhysicsShapeQueryResult.cs ObjectType\PinJoint.cs ObjectType\PinJoint2D.cs ObjectType\PlaneMesh.cs ObjectType\PlaneShape.cs ObjectType\PluginScript.cs ObjectType\Polygon2D.cs ObjectType\PolygonPathFinder.cs ObjectType\Popup.cs ObjectType\PopupDialog.cs ObjectType\PopupMenu.cs ObjectType\PopupPanel.cs ObjectType\Position2D.cs ObjectType\Position3D.cs ObjectType\PrimitiveMesh.cs ObjectType\PrismMesh.cs ObjectType\ProceduralSky.cs ObjectType\ProgressBar.cs ObjectType\ProjectSettings.cs ObjectType\ProximityGroup.cs ObjectType\ProxyTexture.cs ObjectType\QuadMesh.cs ObjectType\Range.cs ObjectType\RayCast.cs ObjectType\RayCast2D.cs ObjectType\RayShape.cs ObjectType\RayShape2D.cs ObjectType\RectangleShape2D.cs ObjectType\Reference.cs ObjectType\ReferenceRect.cs ObjectType\ReflectionProbe.cs ObjectType\RegEx.cs ObjectType\RegExMatch.cs ObjectType\RemoteTransform.cs ObjectType\RemoteTransform2D.cs ObjectType\Resource.cs ObjectType\ResourceInteractiveLoader.cs ObjectType\ResourceLoader.cs ObjectType\ResourcePreloader.cs ObjectType\ResourceSaver.cs ObjectType\RichTextLabel.cs ObjectType\RigidBody.cs ObjectType\RigidBody2D.cs ObjectType\RootMotionView.cs ObjectType\SceneState.cs ObjectType\SceneTree.cs ObjectType\SceneTreeTimer.cs ObjectType\Script.cs ObjectType\ScrollBar.cs ObjectType\ScrollContainer.cs ObjectType\SegmentShape2D.cs ObjectType\Semaphore.cs ObjectType\Separator.cs ObjectType\Shader.cs ObjectType\ShaderMaterial.cs ObjectType\Shape.cs ObjectType\Shape2D.cs ObjectType\ShortCut.cs ObjectType\Skeleton.cs ObjectType\Skeleton2D.cs ObjectType\SkeletonIK.cs ObjectType\Sky.cs ObjectType\Slider.cs ObjectType\SliderJoint.cs ObjectType\SoftBody.cs ObjectType\Spatial.cs ObjectType\SpatialGizmo.cs ObjectType\SpatialMaterial.cs ObjectType\SpatialVelocityTracker.cs ObjectType\SphereMesh.cs ObjectType\SphereShape.cs ObjectType\SpinBox.cs ObjectType\SplitContainer.cs ObjectType\SpotLight.cs ObjectType\SpringArm.cs ObjectType\Sprite.cs ObjectType\Sprite3D.cs ObjectType\SpriteBase3D.cs ObjectType\SpriteFrames.cs ObjectType\StaticBody.cs ObjectType\StaticBody2D.cs ObjectType\StreamPeer.cs ObjectType\StreamPeerBuffer.cs ObjectType\StreamPeerGDNative.cs ObjectType\StreamPeerSSL.cs ObjectType\StreamPeerTCP.cs ObjectType\StreamTexture.cs ObjectType\StyleBox.cs ObjectType\StyleBoxEmpty.cs ObjectType\StyleBoxFlat.cs ObjectType\StyleBoxLine.cs ObjectType\StyleBoxTexture.cs ObjectType\SurfaceTool.cs ObjectType\TabContainer.cs ObjectType\Tabs.cs ObjectType\TCP_Server.cs ObjectType\TextEdit.cs ObjectType\TextFile.cs ObjectType\Texture.cs ObjectType\Texture3D.cs ObjectType\TextureArray.cs ObjectType\TextureButton.cs ObjectType\TextureLayered.cs ObjectType\TextureProgress.cs ObjectType\TextureRect.cs ObjectType\Theme.cs ObjectType\Thread.cs ObjectType\TileMap.cs ObjectType\TileSet.cs ObjectType\Timer.cs ObjectType\ToolButton.cs ObjectType\TouchScreenButton.cs ObjectType\Translation.cs ObjectType\TranslationServer.cs ObjectType\Tree.cs ObjectType\TreeItem.cs ObjectType\TriangleMesh.cs ObjectType\Tween.cs ObjectType\UndoRedo.cs ObjectType\UPNP.cs ObjectType\UPNPDevice.cs ObjectType\VBoxContainer.cs ObjectType\VehicleBody.cs ObjectType\VehicleWheel.cs ObjectType\VideoPlayer.cs ObjectType\VideoStream.cs ObjectType\VideoStreamTheora.cs ObjectType\VideoStreamWebm.cs ObjectType\Viewport.cs ObjectType\ViewportContainer.cs ObjectType\ViewportTexture.cs ObjectType\VisibilityEnabler.cs ObjectType\VisibilityEnabler2D.cs ObjectType\VisibilityNotifier.cs ObjectType\VisibilityNotifier2D.cs ObjectType\VisualInstance.cs ObjectType\VisualScript.cs ObjectType\VisualScriptBasicTypeConstant.cs ObjectType\VisualScriptBuiltinFunc.cs ObjectType\VisualScriptClassConstant.cs ObjectType\VisualScriptComment.cs ObjectType\VisualScriptCondition.cs ObjectType\VisualScriptConstant.cs ObjectType\VisualScriptConstructor.cs ObjectType\VisualScriptCustomNode.cs ObjectType\VisualScriptDeconstruct.cs ObjectType\VisualScriptEmitSignal.cs ObjectType\VisualScriptEngineSingleton.cs ObjectType\VisualScriptExpression.cs ObjectType\VisualScriptFunction.cs ObjectType\VisualScriptFunctionCall.cs ObjectType\VisualScriptFunctionState.cs ObjectType\VisualScriptGlobalConstant.cs ObjectType\VisualScriptIndexGet.cs ObjectType\VisualScriptIndexSet.cs ObjectType\VisualScriptInputAction.cs ObjectType\VisualScriptIterator.cs ObjectType\VisualScriptLocalVar.cs ObjectType\VisualScriptLocalVarSet.cs ObjectType\VisualScriptMathConstant.cs ObjectType\VisualScriptNode.cs ObjectType\VisualScriptOperator.cs ObjectType\VisualScriptPreload.cs ObjectType\VisualScriptPropertyGet.cs ObjectType\VisualScriptPropertySet.cs ObjectType\VisualScriptResourcePath.cs ObjectType\VisualScriptReturn.cs ObjectType\VisualScriptSceneNode.cs ObjectType\VisualScriptSceneTree.cs ObjectType\VisualScriptSelect.cs ObjectType\VisualScriptSelf.cs ObjectType\VisualScriptSequence.cs ObjectType\VisualScriptSubCall.cs ObjectType\VisualScriptSwitch.cs ObjectType\VisualScriptTypeCast.cs ObjectType\VisualScriptVariableGet.cs ObjectType\VisualScriptVariableSet.cs ObjectType\VisualScriptWhile.cs ObjectType\VisualScriptYield.cs ObjectType\VisualScriptYieldSignal.cs ObjectType\VisualServer.cs ObjectType\VisualShader.cs ObjectType\VisualShaderNode.cs ObjectType\VisualShaderNodeColorConstant.cs ObjectType\VisualShaderNodeColorOp.cs ObjectType\VisualShaderNodeColorUniform.cs ObjectType\VisualShaderNodeCubeMap.cs ObjectType\VisualShaderNodeCubeMapUniform.cs ObjectType\VisualShaderNodeDotProduct.cs ObjectType\VisualShaderNodeInput.cs ObjectType\VisualShaderNodeOutput.cs ObjectType\VisualShaderNodeScalarConstant.cs ObjectType\VisualShaderNodeScalarFunc.cs ObjectType\VisualShaderNodeScalarInterp.cs ObjectType\VisualShaderNodeScalarOp.cs ObjectType\VisualShaderNodeScalarUniform.cs ObjectType\VisualShaderNodeTexture.cs ObjectType\VisualShaderNodeTextureUniform.cs ObjectType\VisualShaderNodeTransformCompose.cs ObjectType\VisualShaderNodeTransformConstant.cs ObjectType\VisualShaderNodeTransformDecompose.cs ObjectType\VisualShaderNodeTransformMult.cs ObjectType\VisualShaderNodeTransformUniform.cs ObjectType\VisualShaderNodeTransformVecMult.cs ObjectType\VisualShaderNodeUniform.cs ObjectType\VisualShaderNodeVec3Constant.cs ObjectType\VisualShaderNodeVec3Uniform.cs ObjectType\VisualShaderNodeVectorCompose.cs ObjectType\VisualShaderNodeVectorDecompose.cs ObjectType\VisualShaderNodeVectorFunc.cs ObjectType\VisualShaderNodeVectorInterp.cs ObjectType\VisualShaderNodeVectorLen.cs ObjectType\VisualShaderNodeVectorOp.cs ObjectType\VScrollBar.cs ObjectType\VSeparator.cs ObjectType\VSlider.cs ObjectType\VSplitContainer.cs ObjectType\WeakRef.cs ObjectType\WebSocketClient.cs ObjectType\WebSocketMultiplayerPeer.cs ObjectType\WebSocketPeer.cs ObjectType\WebSocketServer.cs ObjectType\WindowDialog.cs ObjectType\World.cs ObjectType\World2D.cs ObjectType\WorldEnvironment.cs ObjectType\XMLParser.cs ObjectType\YSort.cs Properties\AssemblyInfo.cs "C:\Users\XXXXXXXXX\AppData\Local\Temp\.NETFramework,Version=v4.5.AssemblyAttributes.cs"
		Csc: Using shared compilation with compiler from directory: D:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Roslyn
		ObjectType\ArrayMesh.cs(60,36): warning CS0108: 'ArrayMesh.BlendShapeMode' hides inherited member 'Mesh.BlendShapeMode'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\ArrayMesh.cs(31,21): warning CS0108: 'ArrayMesh.ArrayFormat' hides inherited member 'Mesh.ArrayFormat'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\ArrayMesh.cs(44,21): warning CS0108: 'ArrayMesh.ArrayType' hides inherited member 'Mesh.ArrayType'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\ARVRPositionalTracker.cs(52,39): warning CS0108: 'ARVRPositionalTracker.GetType()' hides inherited member 'object.GetType()'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\CPUParticles.cs(680,22): warning CS0108: 'CPUParticles.Scale' hides inherited member 'Spatial.Scale'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\CPUParticles.cs(9,21): warning CS0108: 'CPUParticles.Flags' hides inherited member 'GeometryInstance.Flags'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\CPUParticles2D.cs(665,22): warning CS0108: 'CPUParticles2D.Scale' hides inherited member 'Node2D.Scale'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\Slider.cs(70,38): warning CS0108: 'Slider.FocusMode' hides inherited member 'Control.FocusMode'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\PathFollow2D.cs(76,21): warning CS0108: 'PathFollow2D.Rotate' hides inherited member 'Node2D.Rotate(float)'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\LightOccluder2D.cs(30,20): warning CS0108: 'LightOccluder2D.LightMask' hides inherited member 'CanvasItem.LightMask'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		ObjectType\LineEdit.cs(140,38): warning CS0108: 'LineEdit.FocusMode' hides inherited member 'Control.FocusMode'. Use the new keyword if hiding was intended. [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		Core\ExtensionsNodeExtensions.cs(7,20): error CS0030: Cannot convert type 'Godot.Node' to 'T' [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		Core\ExtensionsNodeExtensions.cs(17,20): error CS0030: Cannot convert type 'Godot.Node' to 'T' [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		Core\GD.cs(70,20): error CS0030: Cannot convert type 'Godot.Resource' to 'T' [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		Core\ExtensionsNodeExtensions.cs(27,20): error CS0030: Cannot convert type 'Godot.Node' to 'T' [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
		Core\ExtensionsNodeExtensions.cs(37,20): error CS0030: Cannot convert type 'Godot.Node' to 'T' [C:\Users\XXXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
	Done building project "GodotSharp.csproj" -- FAILED.
Done building project "GodotSharp.sln" -- FAILED.
```
</details>


```
Core\ExtensionsNodeExtensions.cs(7,20): error CS0030: Cannot convert type 'Godot.Node' to 'T' [C:\Users\XXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
Core\ExtensionsNodeExtensions.cs(17,20): error CS0030: Cannot convert type 'Godot.Node' to 'T' [C:\Users\XXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
Core\GD.cs(70,20): error CS0030: Cannot convert type 'Godot.Resource' to 'T' [C:\Users\XXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
Core\ExtensionsNodeExtensions.cs(27,20): error CS0030: Cannot convert type 'Godot.Node' to 'T' [C:\Users\XXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
Core\ExtensionsNodeExtensions.cs(37,20): error CS0030: Cannot convert type 'Godot.Node' to 'T' [C:\Users\XXXXXXXX\AppData\Roaming\Godot\mono\solutions\10008632168450808833_3_1538749676\GodotSharp\GodotSharp.csproj]
```

**Steps to reproduce:**
1. Create new project
2. Try to create a c# solution from the top right mono icon
3. Watch errors pile up

**Minimal reproduction project:**
<!-- Recommended as it greatly speeds up debugging. Drag and drop a zip archive to upload it. -->

--- a/modules/mono/glue/Managed/Files/Extensions/NodeExtensions.cs
+++ b/modules/mono/glue/Managed/Files/Extensions/NodeExtensions.cs
@@ -4,7 +4,7 @@ namespace Godot
     {
         public T GetNode<T>(NodePath path) where T : class
         {
-            return (T)GetNode(path);
+            return (T)(object)GetNode(path);
         }
 
         public T GetNodeOrNull<T>(NodePath path) where T : class
@@ -14,7 +14,7 @@ namespace Godot
 
         public T GetChild<T>(int idx) where T : class
         {
-            return (T)GetChild(idx);
+            return (T)(object)GetChild(idx);
         }
 
         public T GetChildOrNull<T>(int idx) where T : class
@@ -24,7 +24,7 @@ namespace Godot
 
         public T GetOwner<T>() where T : class
         {
-            return (T)GetOwner();
+            return (T)(object)GetOwner();
         }
 
         public T GetOwnerOrNull<T>() where T : class
@@ -34,7 +34,7 @@ namespace Godot
 
         public T GetParent<T>() where T : class
         {
-            return (T)GetParent();
+            return (T)(object)GetParent();
         }
 
         public T GetParentOrNull<T>() where T : class

--- a/modules/mono/glue/Managed/Files/Extensions/ResourceLoaderExtensions.cs
+++ b/modules/mono/glue/Managed/Files/Extensions/ResourceLoaderExtensions.cs
@@ -2,9 +2,9 @@ namespace Godot
 {
     public static partial class ResourceLoader
     {
-        public static T Load<T>(string path) where T : Godot.Resource
+        public static T Load<T>(string path) where T : class
         {
-            return (T) Load(path);
+            return (T)(object)Load(path);
         }
     }
 }

--- a/modules/mono/glue/Managed/Files/GD.cs
+++ b/modules/mono/glue/Managed/Files/GD.cs
@@ -67,7 +67,7 @@ namespace Godot
 
         public static T Load<T>(string path) where T : class
         {
-            return (T) ResourceLoader.Load(path);
+            return ResourceLoader.Load<T>(path);
         }
 
         public static void Print(params object[] what)

